### PR TITLE
Handle RTD enabling addons

### LINKED ISSUE #3064: Make sure trio works with Read the Docs Addons
I saw https://about.readthedocs.com/blog/2024/07/addons-by-default/ and it looks like we might have to make some changes. Its not very clear to me what changes exactly though. And it seems the theme support for it is in a dev release so we should probably wait.

### PR DESCRIPTION
**Pull request by A5rocks**
Fixes #3064

I'm not sure whether this is required because the docs look fine, but then of course the docs have very many hacks that I'm not aware of.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -374,10 +381,8 @@ def add_mapping(
 # We have to set this ourselves, not only because it's useful for local
 # testing, but also because if we don't then RTD will throw away our
 # html_theme_options.
-import sphinx_rtd_theme
```
Comment (jakkdl) [inline docs/source/conf.py:384]:
```suggestion
```

let's attach the comment to the line it's talking about

**Review by jakkdl — APPROVED** [review]
seems to work? The generated docs looks normal at least.

**Comment by A5rocks** [comment]
OK I don't know why RTD is failing to clone this PR now...

**Comment by A5rocks** [comment]
Well it seems to work now. I'll merge this soon enough unless someone has comments.

**Review by CoolCat467 — APPROVED** [review]
Looks good

**Pull request merged**